sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsActions.java

```diff
@@ -145,9 +160,19 @@ public TextAnalyticsActions setExtractKeyPhrasesActions(ExtractKeyPhrasesAction.
      * @param analyzeSentimentActions the list of {@link AnalyzeSentimentAction} to be executed.
      *
      * @return the {@link TextAnalyticsActions} object itself.
+     *
+     * @throws IllegalArgumentException if duplicate {@link AnalyzeSentimentAction} actions are passed.
      */
     public TextAnalyticsActions setAnalyzeSentimentActions(AnalyzeSentimentAction... analyzeSentimentActions) {
+        validateActionsNumber(analyzeSentimentActions, AnalyzeSentimentAction.class.getName());
         this.analyzeSentimentActions = analyzeSentimentActions == null ? null : Arrays.asList(analyzeSentimentActions);
         return this;
     }
+
+    private void validateActionsNumber(Object[] actions, String actionType) {
+        if (actions != null && actions.length > 1) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(String.format(
+                "Currently, the service can accept up to one %s action per action type only", actionType)));
```

Comment:
nit: Since you're passing in the name of the action, I think it might sound better to say something like:

```suggestion
                "Currently, the service can accept up to one %s. Duplicate actions not supported.", actionType)));
```

but feel free to leave as is